src/units/abilities.cpp

```diff
@@ -2576,15 +2576,16 @@ effect::effect(const unit_ability_list& list, int def, const const_attack_ptr& a
 		effect_list_.push_back(val.second);
 	}
 
-	composite_value_ = static_cast<int>((value_set + addition + substraction) * multiplier / divisor);
+	composite_double_value_ = (value_set + addition + substraction) * multiplier / divisor;
 	//clamp what if min_value < max_value or one attribute only used.
 	if(max_value && min_value && *min_value < *max_value) {
-		composite_value_ = std::clamp(*min_value, *max_value, composite_value_);
+		composite_double_value_ = std::clamp(*min_value, *max_value, composite_double_value_);
 	} else if(max_value && !min_value) {
-		composite_value_ = std::min(*max_value, composite_value_);
+		composite_double_value_ = std::min(*max_value, composite_double_value_);
 	} else if(min_value && !max_value) {
-		composite_value_ = std::max(*min_value, composite_value_);
+		composite_double_value_ = std::max(*min_value, composite_double_value_);
 	}
+	composite_value_ = std::round(composite_double_value_);
```

Comment:
What logic still uses the `int` composite value? Not sure if it's good to have both. That seems confusing.

Reply:
chance_to_hit attack etc use only int value without calculation after, it for don't have to add round each time what i done like that.

Reply:
@soliton- damage calculation is the only case where the value returned by [damage] is then recalculated by the Tod and [leadership], the attack values ​​chance_to_hit [heals] etc. are used as is and must therefore be integers, [damage] returning a double value is an exception (which personally I could have done without but hey...)

Reply:
> which personally I could have done without but hey...

Why did you write this PR then? Why not suggest a different solution? Or explain how the status quo is just as it should be?

Reply:
the title of issue was:"[damage] weapon special produces unexpected damage values", I understood that it was the value returned by [damage] that had to be rounded to the nearest integer, not the final value of the damage inflicted.